RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Function.cs

```diff
@@ -0,0 +1,139 @@
+                    //The SNS topics are differentiated by char type in the SNS topics name 'ENV-rtl-divNNN-v4-ctNN' 
+                    //This makes it unnecessary to extract the char type id from the attributes, however we are
+                    //showed how this could be done if a customer were to use a generic lambda for multiple SNS topics.
+                    //
+                    var charType = (CharTypeID)(Convert.ToInt32(record.Sns.MessageAttributes[EntityBaseMessageAttributes.CharTypeID].Value));
+                    switch (charType)
+                    {
+                        case CharTypeID.CatalogItem:
+                            Console.WriteLine($"Catalog-Item message received.");
+                            break;
+
+                        case CharTypeID.Rightset:
+                            Console.WriteLine($"Rightset message received.");
+                            break;
+                    }
+
+
+                    //
+                    //act on a particular action as needed
+                    //
+                    var action = record.Sns.MessageAttributes[EntityBaseMessageAttributes.Action].Value;
+                    switch (action)
+                    {
+                        case EntityBaseMessageActions.EntityActionCreated:
+                            Console.WriteLine($"Created");
+                            break;
+
+                        case EntityBaseMessageActions.EntityActionUpdated:
+                            Console.WriteLine($"Updated");
+                            break;
+
+                        case EntityBaseMessageActions.EntityActionDeleted:
+                            Console.WriteLine($"Deleted");
+                            break;
+                    }
+
+                    var messageEntity = JsonConvert.DeserializeObject<ModuleEntityMessage>(record.Sns.Message, Converter.Settings);
+
+                    //SAMPLES
+                    //Get a catalog item - assuming it was the message received on the queue
+                    var catalogResult = await this._v4.Get("catalog-item", messageEntity.Entity.EntityId);
+
+                    //Search for the first 25 catalog items with the same template as the one received on the queue
+                    var catalogPayload = @"
+                        {""query"":
+                        {""$and"":[
+                        {""$eq"":[""templateid""," + messageEntity.Entity.Template.TemplateId + @"]}]},
+                        ""start"":0,""rows"":25,""sortOrders"":[""sequencenumber asc""]}";
+
+                    var catalogResults = await this._v4.Search("catalog-item", catalogPayload);
+
+                    if (catalogResults.Count() == 0)
+                    {
+                        Console.WriteLine($"Cannot find Catalog Items with template id {messageEntity.Entity.Template.TemplateId}.");
+                        Console.WriteLine("Lambda processing aborted");
+                    }
+
```

Comment:
maybe add an example of grabbing additional information from a rightset in here also? or getting all catalog items associated to a rightset when it is changed?